modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/dag/schema.scala

```diff
@@ -11,6 +11,24 @@ object schema {
 
   case class GenericMetadata()
 
+  @derive(eqv, show)
+  sealed trait SignatureType extends EnumEntry
+
+  object SignatureType extends Enum[SignatureType] with SignatureTypeCodecs {
```

Comment:
It looks duplicated (with a case class)

Reply:
`SignatureType` is a duplicate of `ChainObjectStatus` from the same file and is meant to be a JSON enum.

Reply:
but there is also SignatureType case class (autogenerated)

Reply:
I get what you are trying to say now. Changed to `SignatureTypeValue`.

Reply:
Why? What does it solve?

Reply:
Do you have a recommendation for what to name this?

Reply:
SignatureType was totally fine name. What I meant was that it was duplicated (both empty case class and trait+object).

Reply:
So, [this](https://github.com/cngo-github/tessellation/blob/20230105-rosetta_classes_pt2/modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/SignatureType.scala) is the auto-generated case class. We are keeping it so that we can update it down the line as the auto-generated Rosetta classes change. [This](https://github.com/cngo-github/tessellation/blob/20230105-rosetta_classes_pt2/modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/dag/schema.scala#L15-L35) is an `enum` meant for this [field](https://github.com/cngo-github/tessellation/blob/20230105-rosetta_classes_pt2/modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/Signature.scala#L13) as specified [here](https://www.rosetta-api.org/docs/models/SignatureType.html).